### PR TITLE
[JIT] clean up

### PR DESCRIPTION
**Pull request by garymm**
* Minor: spelling, grammar.
* Add calls to `GRAPH_DUMP()` where they were missing.
* Add or expand a few comments.
* Move a few comments to seemingly more appropriate spots.
* In canonicalize_graph_fuser_ops.cpp inline `runnableInputs()` since it
  was only called in one place and had a misleading comment and
  confusing name.
* In `PeepholeOptimizeImpl::optimizeBlock()`, set `changed = true;` when
  removing `aten::is_complex`. Pretty sure its absence was a bug.
* Delete unused `_jit_pass_remove_inplace_ops` and and its
  implementation `RemoveInplaceOps()`.
* In `preprocessCaffe2Ops()`, remove redundant check for nested optional
  types. It was already checked in `checkONNXCompatibility()`.
* In `EncoderBase::AddAttribute`, log the unexpected attribute kind.
  I don't remember the repro case now but I did hit this error at some
  point and this additional logging made it easier to understand.
* In `fuseConvBatchNorm()` in eval_peephole.cpp, consistently use
  camelCase instead of snake_case for local variables.
* Add curly braces around the bodies of if and loops.

### CODE REVIEW

File: torch/csrc/jit/passes/onnx/eval_peephole.cpp [hunk 1]
```diff
@@ -111,21 +112,21 @@ static void fuseConvBatchNorm(Block* b, ValueToParamPairMap& valsToParamsMap) {
...
       }
 
-      convNode->copyAttributes(*origconvNode);
+      convNode->copyAttributes(*origConvNode);
       convNode->insertBefore(bnNode);
-      convNode->addInput(origconvNode->inputs().at(0));
+      convNode->addInput(origConvNode->inputs().at(0));
 
-      auto conv_W = b->owningGraph()->addInput();
+      auto convW = b->owningGraph()->addInput();
```
Comment (neginraoof) [inline torch/csrc/jit/passes/onnx/eval_peephole.cpp:119]: Suggestion: now that we are reformatting the names, can we also rename the params in this pass to have more helpful names?

File: torch/csrc/jit/passes/remove_inplace_ops.cpp [hunk 2]
```diff
@@ -85,18 +12,19 @@ void RemoveInplaceOps(Block* block) {
...
 // After:
-// graph(%0 : Float),
-//        %1 : Half):
-//   %4 : Half = onnx::Cast[to=10](%0)
-//   %5 : Half = onnx::Add(%1, %4)
+// graph(%0 : Half),
+//       %1 : Float):
+//   %2 : Half = aten::type_as(%1, %0)
+//   # Half + Half will result in correct dtype.
+//   : = aten::add_(%0, %2)
```
Comment (neginraoof) [inline torch/csrc/jit/passes/remove_inplace_ops.cpp:26]: Is the type Half missing here?

File: torch/csrc/jit/passes/remove_inplace_ops.cpp [hunk 3]
```diff
@@ -85,18 +12,19 @@ void RemoveInplaceOps(Block* block) {
...
 // Before:
-// graph(%0 : Float),
-//        %1 : Half):
-//   %4 : Float = onnx::Cast[to=1](%1)
-//   %5 : Float = onnx::Add(%4, %0)
+// graph(%0 : Half),
+//       %1 : Float):
+//   # Should result in a Half, but after translation to out-of-place,
+//   # would become a Float b/c Half+Float -> Float.
+//   : = aten::add_(%0, %1)
```
Comment (neginraoof) [inline torch/csrc/jit/passes/remove_inplace_ops.cpp:19]: Same, is the type missing?

**Review by neginraoof — COMMENTED** [review]
Thanks for the update, left a few comments.

File: torch/csrc/jit/passes/onnx/eval_peephole.cpp (hunk 1 above)
Comment (garymm) [reply torch/csrc/jit/passes/onnx/eval_peephole.cpp]: Tried to improve the names and add some comments that hopefully make the meaning clearer.

File: torch/csrc/jit/passes/remove_inplace_ops.cpp (hunk 3 above)
Comment (garymm) [reply torch/csrc/jit/passes/remove_inplace_ops.cpp]: Done.

File: torch/csrc/jit/passes/remove_inplace_ops.cpp (hunk 2 above)
Comment (garymm) [reply torch/csrc/jit/passes/remove_inplace_ops.cpp]: Done.

**Review by garymm — COMMENTED** [review]
Thanks for the review. Please take another look and approve if you think this is ready for review by FB.

**Review by neginraoof — APPROVED** [review]
Thanks!

**Comment by garymm** [comment]
@SplitInfinity could you please review and import?

**Review by SplitInfinity — APPROVED** [review]

File: torch/csrc/jit/passes/onnx.cpp [hunk 4]
```diff
@@ -91,14 +93,9 @@ void preprocessCaffe2Ops(Block* block) {
         auto type = arg.type();
         AT_ASSERT(origin_inputs_index < origin_inputs.size());
         const auto& origin_input = origin_inputs[origin_inputs_index++];
-        if (type->kind() == TypeKind::OptionalType) {
-          type = reinterpret_cast<OptionalType*>(type.get())->getElementType();
-          if (origin_input->mustBeNone()) {
-            continue;
-          } else {
-            // recursive optional type is not supported
-            AT_ASSERT(type->kind() != TypeKind::OptionalType);
```
[the remark this answers is not in the document]
Comment (garymm) [reply torch/csrc/jit/passes/onnx.cpp:100]: I'm pretty sure this is asserted in checkONNXCompatibility, which is called above, so this is guaranteed to always be true on this line (this is mentioned in the commit message, but let me know if the wording wasn't clear).

File: torch/csrc/jit/passes/onnx/constant_fold.cpp [hunk 5]
```diff
@@ -621,10 +620,10 @@ void ConstantFoldONNX(Block* b, ParamMap& paramsDict, int opset_version) {
     // Next we remove the current node that has been replaced by
     // an initializer. But before we start de-wiring this node,
     // we check if any parents of this nodes were onnx::Constant
-    // and remove them first (following proper sequence as shown
-    // below), and then remove the current node. If the parent was
-    // an initializer (not onnx::Constant) then they are all removed
-    // by eraseUnusedBlockInputs() call (below) outside the loop.
+    // and remove them first, and then remove the current node.
+    // If the parent was an initializer (not onnx::Constant) then
+    // they are all removed by eraseUnusedBlockInputs() call (below)
```
[the remark this answers is not in the document]
Comment (garymm) [reply torch/csrc/jit/passes/onnx/constant_fold.cpp]: Done.

**Comment by garymm** [comment]
It seems clang-tidy is failing [because it can't find omp.h](https://github.com/pytorch/pytorch/pull/60390/checks?check_run_id=2956011342#step:6:149).
I reproduced locally and then fixed it by installing openmp.
No idea how to fix this in CI though. Any suggestions?

**Comment by garymm** [comment]
@SplitInfinity when you have a chance please let me know if there's a problem with the internal tests:
https://github.com/pytorch/pytorch/pull/60390/checks?check_run_id=2966019584

The other check failures seem unrelated to this change.

**Pull request closed without merging**